core/src/main/java/bisq/core/user/Preferences.java

```diff
@@ -692,10 +692,12 @@ public boolean showAgain(String key) {
     }
 
     public boolean getUseTorForBitcoinJ() {
-        // We override the useTorForBitcoinJ and set it to false if we detected a localhost node or if we are not on mainnet.
-        // At testnet there are very few Bitcoin tor nodes and we don't provide tor nodes.
-        if (!BisqEnvironment.getBaseCurrencyNetwork().isMainnet()
+        // We override the useTorForBitcoinJ and set it to false if we detected a localhost node or if we are not on mainnet,
+        // unless the useTorForBtc parameter is explicitly provided.
+        // On testnet there are very few Bitcoin tor nodes and we don't provide tor nodes.
+        if ((!BisqEnvironment.getBaseCurrencyNetwork().isMainnet()
                 || bisqEnvironment.isBitcoinLocalhostNodeRunning())
+                && bisqEnvironment.getProperty(BtcOptionKeys.USE_TOR_FOR_BTC).isEmpty())
```

Comment:
Good point. I will rework that.

Reply:
After reviewing the code again, my initial implementation appears to be technically correct and attempts to determine if the USE_TOR_FOR_BTC parameter was explicitly provided or not by checking whether the property is empty (which it will be empty if not provided). And if it is provided, do not override it to false and instead use whatever value was provided. It handles each scenario where (1) the parameter is not provided, (2) provided and set to false, and (3) provided and set to true. I did not change the implementation/type of the USE_TOR_FOR_BTC as a string parameter to boolean in order to keep it consistent with other similar parameters. If that needs to change, that can be done separately. The only thing I identified was that I should have instead used the useTorFlagFromOptions parameter rather than bisqEnvironment.getProperty. I will commit a fix for that.